### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by quietbits**
`error.toString()` adds "Error: " prefix to all error messages, `error.message` makes it cleaner.

### CODE REVIEW

File: src/components/AddAsset.tsx [hunk 1]
```diff
@@ -56,8 +56,8 @@ export const AddAsset = ({ onClose }: { onClose: () => void }) => {
       );
       setLocalStatus(ActionStatus.SUCCESS);
     } catch (e) {
-      log.error({ title: e.toString() });
-      setErrorMessage(e.toString());
+      log.error({ title: e.message });
```
Comment (quietbits) [inline src/components/AddAsset.tsx:59]: @piyalbasu Do you think we should make a helper function to display `error.message` or `error.toString()` if `message` is not available. Errors that we are throwing should have a `message` prop, but there might be cases where it comes from somewhere else and doesn't have it? 🤔

Comment (piyalbasu) [reply src/components/AddAsset.tsx:59]: I think that's a good idea! I'm running into a similar issue in Freighter where `e` doesn't always have a `message` attached to it

File: src/helpers/getValidatedUntrustedAsset.ts [hunk 2]
```diff
@@ -32,8 +33,8 @@ export const getValidatedUntrustedAsset = async ({
 
     asset = `${assetCode}:${homeDomainIssuer}`;
   } catch (e) {
-    log.error({ title: "Issuer domain error: ", body: e.message });
-    throw new Error(e.message);
+    log.error({ title: "Issuer domain error: ", body: getErrorMessage(e) });
+    throw new Error(e);
```
Comment (piyalbasu) [inline src/helpers/getValidatedUntrustedAsset.ts:37]: Should you be passing Error a string? `throw new Error(getErrorMessage(e))`

Not sure what kind of errors `getIssuerFromDomain` might throw, but you might end up in wrapping something that's already an Error object in another Error object

File: src/components/AddAsset.tsx [hunk 3]
```diff
@@ -56,8 +57,8 @@ export const AddAsset = ({ onClose }: { onClose: () => void }) => {
       );
       setLocalStatus(ActionStatus.SUCCESS);
     } catch (e) {
-      log.error({ title: e.message });
-      setErrorMessage(e.message);
+      log.error({ title: getErrorMessage(e) });
+      setErrorMessage(getErrorMessage(e));
       setLocalStatus(ActionStatus.ERROR);
```
Comment (piyalbasu) [inline src/components/AddAsset.tsx:62]: very very small, but probably best practice to cache the value of `getErrorMessage(error)` as a const in instances like this and below where the value isn't changing

File: src/ducks/sep24DepositAsset.ts [hunk 4]
```diff
@@ -126,11 +127,11 @@ export const depositAssetAction = createAsyncThunk<
     } catch (error) {
       log.error({
         title: "Deposit failed",
-        body: error.message,
+        body: getErrorMessage(error),
       });
 
       return rejectWithValue({
-        errorString: error.message,
+        errorString: getErrorMessage(error),
```
Comment (piyalbasu) [inline src/ducks/sep24DepositAsset.ts:134]: same as above - probably best to cache error when you're logging and then rejecting with the same string

**Review by piyalbasu — APPROVED** [review]

File: src/helpers/getValidatedUntrustedAsset.ts (hunk 2 above)
Comment (quietbits) [reply src/helpers/getValidatedUntrustedAsset.ts:37]: 👍 You're right, I'll fix that. We have more control if we pass along the error string.

**Pull request merged**